src/App.tsx

```diff
@@ -2,24 +2,52 @@ import { Box, ChakraProvider } from '@chakra-ui/react';
 import React, { useEffect, useMemo } from 'react';
 import { useActor, useInterpret, useSelector } from '@xstate/react';
 import { useAuth } from './authContext';
+import { AppHead } from './AppHead';
 import { CanvasProvider } from './CanvasContext';
-import { EmbedProvider, useEmbed } from './embedContext';
+import { EmbedProvider } from './embedContext';
 import { CanvasView } from './CanvasView';
-import './Graph';
 import { isOnClientSide } from './isOnClientSide';
 import { MachineNameChooserModal } from './MachineNameChooserModal';
 import { PaletteProvider } from './PaletteContext';
 import { paletteMachine } from './paletteMachine';
 import { PanelsView } from './PanelsView';
 import { SimulationProvider } from './SimulationContext';
 import { simulationMachine } from './simulationMachine';
-import { getSourceActor } from './sourceMachine';
+import { getSourceActor, useSourceRegistryData } from './sourceMachine';
 import { theme } from './theme';
 import { EditorThemeProvider } from './themeContext';
 import { EmbedContext, EmbedMode } from './types';
 import { useInterpretCanvas } from './useInterpretCanvas';
 import { useRouter } from 'next/router';
 import { parseEmbedQuery, withoutEmbedQueryParams } from './utils';
+import { registryLinks } from './registryLinks';
+
+const defaultHeadProps = {
+  title: 'XState Visualizer',
+  ogTitle: 'XState Visualizer',
+  description: 'Visualizer for XState state machines and statecharts',
+  // TODO - get an OG image for the home page
+  ogImageUrl: null,
+};
+
+const VizHead = () => {
+  const sourceRegistryData = useSourceRegistryData();
```

Comment:
This allows us to keep `<head/>` in sync with in-app data despite the app not being remounted (previously it was in sync because the whole app was being remounted)